### ISSUE TITLE
Mysql queryNodeStream

### ISSUE THREAD

**Issue by crisward** [ForbesLindesay/atdatabases#171]
Just a quick question. Does `queryNodeStream` kill the query if the connection is closed?

If not, is their a way to get the thread id, so I can run ```db.query(sql`KILL ${thread_id}`)``` ?

**Comment by ForbesLindesay** [comment]
I'm not sure. The streaming APIs are very difficult to test. If you can produce a test case showing a problem where I just need a MySQL docker container and a JavaScript file to run, I am happy to look into fixing issues.

**Comment by crisward** [comment]
Fair enough... is this something you'd expect your library to deal with, or something that should be handled my the "mysql2" library? I found this issue there https://github.com/sidorares/node-mysql2/issues/664 (from 2017).

**Comment by ForbesLindesay** [comment]
I would expect my library to deal with this, but I suspect it won't. In general my policy with @databases is that I try to provide a secure, easy to use & reliable API, regardless of the underlying libraries. The underlying libraries deal with a lot of low level protocol challenges that make it harder for them to find time for developer conveniences.

**Comment by crisward** [comment]
Ok.

So adding something here would make sense ?
https://github.com/ForbesLindesay/atdatabases/blob/master/packages/mysql/src/__tests__/stream.test.mysql.ts

We'd need to to run a slowish streamed query, then close the stream and expect the database to kill the query. May be able to do that without using a big table. Perhaps using eg `select sleep(1);` to wait 1 second.

We should also be able to find out if the process was killed by listing killed queries with `Select concat('KILL ',id,';') from information_schema.processlist`. Though I'd also need to know the id for the process id prior to closing the connection. Actually counting the processes before and after maybe enough, as it should loose one after the kill.

I'll take a look, though it probably won't be today. 

I can probably do a pull request for the implementation too.

**Comment by ForbesLindesay** [comment]
That sounds great.
